mmv1/third_party/terraform/provider/provider_internal_test.go

```diff
@@ -688,3 +688,320 @@ func TestProvider_providerConfigure_project(t *testing.T) {
+		ConfigValues        map[string]interface{}
+		EnvVariables        map[string]string
+		ExpectError         bool
+		ExpectFieldUnset    bool
+		ExpectedSchemaValue string
+		ExpectedConfigValue string
+	}{
+		"when request_reason is unset in the config, environment variable CLOUDSDK_CORE_REQUEST_REASON is used": {
+			ConfigValues: map[string]interface{}{
+				// request_reason unset
+				"credentials": transport_tpg.TestFakeCredentialsPath,
+			},
+			EnvVariables: map[string]string{
+				"CLOUDSDK_CORE_REQUEST_REASON": "test",
+			},
+			ExpectedSchemaValue: "test",
+			ExpectedConfigValue: "test",
+		},
+		"request_reason set in the config is not overridden by environment variables": {
+			ConfigValues: map[string]interface{}{
+				"request_reason": "request test",
+				"credentials":    transport_tpg.TestFakeCredentialsPath,
+			},
+			EnvVariables: map[string]string{
+				"CLOUDSDK_CORE_REQUEST_REASON": "test",
+			},
+			ExpectedSchemaValue: "request test",
+			ExpectedConfigValue: "request test",
+		},
+		"when no request_reason is provided via config or environment variables, the field remains unset without error": {
+			ConfigValues: map[string]interface{}{
+				// request_reason unset
+				"credentials": transport_tpg.TestFakeCredentialsPath,
+			},
+			ExpectedConfigValue: "",
+		},
+	}
+
+	for tn, tc := range cases {
+		t.Run(tn, func(t *testing.T) {
+
+			// Arrange
+			ctx := context.Background()
+			unsetTestProviderConfigEnvs(t)
+			setupTestEnvs(t, tc.EnvVariables)
+			p := Provider()
+			d := tpgresource.SetupTestResourceDataFromConfigMap(t, p.Schema, tc.ConfigValues)
+
+			// Act
+			c, diags := providerConfigure(ctx, d, p)
+
+			// Assert
+			if diags.HasError() && !tc.ExpectError {
+				t.Fatalf("unexpected error(s): %#v", diags)
+			}
+			if !diags.HasError() && tc.ExpectError {
+				t.Fatal("expected error(s) but got none")
+			}
+			if diags.HasError() && tc.ExpectError {
+				v, ok := d.GetOk("request_reason")
+				if ok {
+					val := v.(string)
+					if val != tc.ExpectedSchemaValue {
+						t.Fatalf("expected request_reason value set in provider data to be %s, got %s", tc.ExpectedSchemaValue, val)
+					}
+					if tc.ExpectFieldUnset {
+						t.Fatalf("expected request_reason value to not be set in provider data, got %s", val)
+					}
+				}
+				// Return early in tests where errors expected
+				return
+			}
+
+			v := d.Get("request_reason")
+			val := v.(string)
+			config := c.(*transport_tpg.Config) // Should be non-nil value, as test cases reaching this point experienced no errors
+
+			if v != tc.ExpectedSchemaValue {
+				t.Fatalf("expected request_reason value set in provider data to be %s, got %s", tc.ExpectedSchemaValue, val)
+			}
+			if config.RequestReason != tc.ExpectedConfigValue {
+				t.Fatalf("expected request_reason value in provider struct to be %s, got %s", tc.ExpectedConfigValue, config.Credentials)
+			}
+		})
+	}
+}
+
+func TestProvider_providerConfigure_batching(t *testing.T) {
```

Comment:
Please add `"credentials": transport_tpg.TestFakeCredentialsPath,` to all the test cases in this test